### PR TITLE
[WIP] Add usage page

### PR DESCRIPTION
**Pull request by brunobat**
Contains the "Telemetry Data Collection Notice".
The link will be used in acceptance question, when running devmode in the console.
Needed for this PR: https://github.com/quarkusio/quarkus/pull/33555

### CODE REVIEW

**Comment by brunobat** [comment]
Code will need changes. Now discussing.

**Comment by paulrobinson** [comment]
Shouldn't the content be in adoc or md?

**Comment by paulrobinson** [comment]
I think it would be good to have a friendly summary at the top that explains the intent of this feature: Engineer talking to an Engineer. Jumping straight into the Legally approved text is a bit overwhelming (Lawyer talking to an Engineer).

I don't think we should hold up the release for this though. Maybe @insectengine can work on a top piece of content that we can follow up with this week or next.

**Comment by brunobat** [comment]
> Shouldn't the content be in adoc or md?

Similar pages are created in plain HTML.

**Comment by paulrobinson** [comment]
LGTM. However, I suspect @maxandersen might want to take a look too.

**Comment by maxandersen** [comment]
We should get the page "engineer friendly" before pushing it into a release though.

**Comment by paulrobinson** [comment]
> > We should get the page "engineer friendly" before pushing it into a release though.
> 
> So you mean before the 3.2.0.Final release? If so, I agree.

@insectengine can you help us create a more engineer friendly version of this page? I I think we need to keep the legal text as is, but we need something that the engineer sees first that explains to them in clearer language what we are doing and why. 

I appreciate you will need more details than this! Maybe we can discuss it tomorrow (Tues) and mock something up for @maxandersen, @brunobat and others to review.

**Comment by maxandersen** [comment]
I pushed an update to make my comment more explicit - still drafty as I just put down the minimal content.

Just to show the idea.

mainly that quarkus.io/usage will show up front the prompt we ask in dev mode. explain it is anonymous and what is used for. Link to separate explicit notice page and then explain how to explicitly turn on/off.

File: usage.md
Comment (brunobat) [inline usage.md:8]: devmode -> dev mode

File: usage.md
Comment (brunobat) [inline usage.md:13]:
```suggestion
If you answer Yes then, when you perform a quarkus build, anonymized data is sent
to gather usage statistics on how Quarkus is being used and adopted.
```

File: usage.md
Comment (brunobat) [inline usage.md:44]:
```suggestion
```
I wouldn't mention this.
This behaviour was not discussed. Explicit false should bypass the User's prompt question and send analytics?

Comment (maxandersen) [reply usage.md:44]: true - i'll delete it.

File: usage.md
Comment (paulrobinson) [inline usage.md:8]: Starting from Quarkus 3.2, when you run Quarkus for the first time in dev mode (`quarkus dev`, `mvn quarkus:dev`, etc.), you are asked if you agree to contribute anonymous build data to the Quarkus community.

Comment (paulrobinson) [inline usage.md:8]:
```suggestion
Starting from Quarkus 3.2, when you run Quarkus for the first time in dev mode (`quarkus dev`, `mvn quarkus:dev`, etc.), you are asked if you agree to contribute anonymous build data to the Quarkus community.
```

File: usage.md
Comment (paulrobinson) [inline usage.md:12]:
```suggestion
If you answer 'Yes', then anonymized data will be sent when you perform a quarkus build,
```

File: usage.md
Comment (paulrobinson) [inline usage.md:13]: I think this shouldn't be on a new line. Not a biggy, but makes it harder to review and propose changes.

File: usage.md
Comment (paulrobinson) [inline usage.md:20]:
```suggestion
gauge which operating systems, java versions, build systems, extensions and more are 
```

File: usage.md
Comment (paulrobinson) [inline usage.md:23]:
```suggestion
We will share what is learned from this anonymous usage data and plan to integrate aggregated forms, such as usage and adoption, 
```

Comment (paulrobinson) [inline usage.md:23]:
```suggestion
We will share what is learned from this anonymous usage data and plan to integrate aggregated forms, such as usage and adoption, 
```

File: usage.md
Comment (paulrobinson) [reply usage.md:44]: Also, I presume you mean "disable it for a single run".

Mentioning this incase you decide not to delete it.

File: _includes/usagepolicy.html
Comment (gsmet) [inline _includes/usagepolicy.html:4]:
```suggestion
      <p>This page contains the policy for the Red Hat sponsored anonymous data collection. See the <a href="/usage">Usage</a> page for details.</a></p>
```

File: _includes/usagepolicy.html
Comment (gsmet) [inline _includes/usagepolicy.html:5]: This paragraph is very worrying. I tried to tone it down a bit. I know we describe precisely what we collect later but you can't be sure people will read it all.
```suggestion
      <p class="intropara">In connection with your use of this Red Hat offering, Red Hat may collect build-time telemetry data about your use of the software. This data allows Red Hat and community extension writers to understand how the software is being consumed and to improve Red Hat offerings and support, including identifying, troubleshooting, and responding to issues that impact users. The data may be shared internally within Red Hat to improve the user experience.</p>
```

File: _includes/usagepolicy.html
Comment (gsmet) [inline _includes/usagepolicy.html:7]: This sentence is weird: `Metrics include the following information such as`. `following` + `such as` doesn't sound good to me.
Also I think we need to be comprehensive as to the nature of the data collected, so I wouldn't use `such as`.

File: _includes/usagepolicy.html
Comment (gsmet) [inline _includes/usagepolicy.html:10]: This can be a bit concerning as errors could contain information about the software developed (a stacktrace for instance can contain private information). I think we need to be precise here.

File: _includes/usagepolicy.html
Comment (gsmet) [inline _includes/usagepolicy.html:15]: Let's not be less precise here than in our documentation: my understanding was we were collecting extensions of specific groupIds.

File: _includes/usagepolicy.html
Comment (gsmet) [inline _includes/usagepolicy.html:18]:
```suggestion
        <li>If it's built with Kubernetes</li>
```

File: _includes/usagepolicy.html
Comment (gsmet) [inline _includes/usagepolicy.html:27]: I thought our data didn't identify users? We need to be consistent in the wording.

File: _includes/usagepolicy.html
Comment (gsmet) [inline _includes/usagepolicy.html:21]: `To the extent that any telemetry data constitutes personal data` <- what exactly does it mean?

File: usage.md
Comment (gsmet) [inline usage.md:3]: Shouldn't the title be `Quarkus Build Analytics`? or whatever we use in the main repo doc?

File: usage_policy.md
Comment (gsmet) [inline usage_policy.md:5]: The image above is outdated I think as we made it point to `usage` (I created a PR and you approved it)?

File: usage.md
Comment (gsmet) [inline usage.md:15]:
```suggestion
This service is provided by Red Hat. More details can be found in the [usage policy](/usage/policy) page.
```

**Review by gsmet — CHANGES_REQUESTED** [review]
I raised several concerns and spotted a few typos.

File: _includes/usagepolicy.html
Comment (maxandersen) [reply _includes/usagepolicy.html:7]: this is how others are defined by legal. we can and should definitely post examples but I think its explicit put here to not require updating it on every small change or adjustment we ne need.

File: _includes/usagepolicy.html
Comment (maxandersen) [reply _includes/usagepolicy.html:10]: We are currently not capturing these so don't have examples to show.

Idea is to be allowed to collect, i.e. the type of exception and obfuscate out any stackframes we don't know - but that's to be added to /usage page once we actually start collecting it (if ever)

File: _includes/usagepolicy.html
Comment (maxandersen) [reply _includes/usagepolicy.html:27]: Yes - but that data is not able to identify users does not mean people with no business justification are permitted to access the resulting data. This is a general data gathering protection requirement IMO.

File: usage_policy.md
Comment (maxandersen) [reply usage_policy.md:5]: yes the image was made before the PR was merged.

/usage is for the general landing page, o that is a link to /usage/policy that has the legal verbage.

File: usage.md
Comment (maxandersen) [reply usage.md:15]: will push fix for this though I thought things written is on the page not in ?

File: _includes/usagepolicy.html
Comment (brunobat) [reply _includes/usagepolicy.html:10]: I think this is outdated. Will remove.

File: _includes/usagepolicy.html
Comment (brunobat) [reply _includes/usagepolicy.html:15]: I think saying this should be enough:
List of public extensions and their versions, included in the Quarkus platform.

File: usage.md
Comment (brunobat) [reply usage.md:3]: What about: `Quarkus Build Analytics Usage` ?

**Comment by brunobat** [comment]
rebased

**Pull request merged**
